packages/explorer/src/hooks/useChain.ts

```diff
@@ -0,0 +1,16 @@
+import { useParams } from "next/navigation";
+import { chainNameId, chains, isValidChainName } from "../common";
+
+export function useChain() {
+  const { chainName } = useParams();
+  isValidChainName(chainName);
+
+  const chainId = chainNameId[chainName];
+  const chain = chains[chainId];
+  return chain;
+}
+
+export function useChainId() {
```

Comment:
Is this hook necessary?

Reply:
it's a nice helper used in `useContractMutation` and such where we need `chainId` only but not strictly necessary. Do you think it's an overkill?

Reply:
yeah I find it a bit overkill when you can do 
```ts
const { id: chainId } = useChain();
```

Reply:
makes sense, took inspiration from wagmi but indeed not necessary